src/net.cpp

```diff
@@ -2884,3 +2883,23 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(const CNode& node, const std::string& msg_type, const Span<const unsigned char>& data, bool is_incoming)
+{
+    auto time = GetTime<std::chrono::microseconds>();
+
+    fs::path base_path = GetDataDir() / "message_logging" / node.addr.ToString();
+    fs::create_directories(base_path);
+
+    fs::path path = base_path / (is_incoming ? "msgs_recv.dat" : "msgs_sent.dat");
```

Comment:
Have you considered including the date in the filename? That would allow for `gzip`:ing files older than say N days via cron. That would be useful for my use cases (see previous comments) at least. The logs will be huge :)

The naming could be something along the lines of `msgs_recv.2020-08-19.dat` and `msgs_sent.2020-08-19.dat`.

You can get the current date in ISO-8601 format using `FormatISO8601Date(GetTime()))` :)